src/Features/CSharp/Portable/MakeLocalFunctionStatic/GetCaptures.cs

```diff
@@ -0,0 +1,189 @@
+using System.Collections.Immutable;
+using System.Composition;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis.CodeGeneration;
+using Microsoft.CodeAnalysis.CSharp.CodeGeneration;
+using Microsoft.CodeAnalysis.CSharp.Extensions;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using Microsoft.CodeAnalysis.Editing;
+using Microsoft.CodeAnalysis.FindSymbols;
+using Microsoft.CodeAnalysis.Host;
+using Microsoft.CodeAnalysis.Host.Mef;
+using Microsoft.CodeAnalysis.PooledObjects;
+using Microsoft.CodeAnalysis.Shared.Extensions;
+using Roslyn.Utilities;
+
+
+
+
+namespace Microsoft.CodeAnalysis.CSharp.GetCapturedVariables
+{
+    [Shared]
+    [ExportLanguageService(typeof(GetCaptures), LanguageNames.CSharp)]
+    internal class GetCaptures : ILanguageService
+    {
+        private static readonly char[] s_underscore = { '_' };
+        private static readonly SyntaxGenerator s_generator = CSharpSyntaxGenerator.Instance;
+
+
+
+        internal async Task<Solution> CreateParameterSymbolAsync(Document document, LocalFunctionStatementSyntax localfunction, CancellationToken cancellationToken)
+        {
+            var semanticModel = await document.GetSemanticModelAsync(cancellationToken).ConfigureAwait(true);
+            if (semanticModel == null)
+            {
+                return document.Project.Solution;
+            }
+
+
+
+            var localFunctionSymbol = semanticModel.GetDeclaredSymbol(localfunction, cancellationToken);
+            var dataFlow = semanticModel.AnalyzeDataFlow(localfunction);
+            var captures = dataFlow.CapturedInside;
+
+
+
+
+            var parameters = DetermineParameters(captures);  
+
+
+
+
+            //Finds all the call sites of the local function
+            var workspace = document.Project.Solution.Workspace;
+            var arrayNode = await SymbolFinder.FindReferencesAsync(localFunctionSymbol, document.Project.Solution, cancellationToken: cancellationToken).ConfigureAwait(false);
+
+
+
+
+            //Initializes a dictionary to replace the nodes of the call sites to be filled with arguments
+            Dictionary<SyntaxNode, SyntaxNode> dict = new Dictionary<SyntaxNode, SyntaxNode>();
+
+
+
+
+            //keep trivia
+            foreach (var referenceSymbol in arrayNode)
+            {
+                foreach (var location in referenceSymbol.Locations)
+                {
+                    var root = await location.Document.GetSyntaxRootAsync(cancellationToken).ConfigureAwait(false);
+                    var syntaxNode = root.FindNode(location.Location.SourceSpan); //Node for the identifier syntax
+
+
+
+
+                    var invocation = (syntaxNode as IdentifierNameSyntax).Parent as InvocationExpressionSyntax;
+                    if (invocation == null)
+                    {
+                        return document.Project.Solution;
+                    }
+
+
+
+                    var arg_List = invocation.ArgumentList;
```

Comment:
variable name doesn't conform to our code style, should use camelCase for locals